src/foam/nanos/auth/AgentUserAuthService.java

```diff
@@ -100,6 +82,45 @@ public User actAs(X x, User entity) throws AuthenticationException {
     return user;
   }
 
+  public boolean canActAs(X x, User agent, User entity) {
+    try {
+      Group group = (Group) groupDAO_.find(entity.getGroup());
+      
+      if ( group == null ) {
+        throw new AuthorizationException("Entity must exist within a group.");
+      }
+      
+      if ( ! group.getEnabled() ) {
```

Comment:
I'm curious about this error case.  Is there any harm in continuing even if the group is disabled?  Presumably the security should be enforced at another level such that a disabled group cannot perform any actions.  Whether or not I can act as a target group should be true regardless of whether that target group is enabled or not, it just so happens that by acting as a disabled group I can't actually perform any actions.

Reply:
Going to ask @zahnno and @TW80000 to take this one

Reply:
I agree with you that we should be able to remove this and everything should still work as expected because we should have many checks in many places that will catch it down the line. I think this was added here just because it seemed like a case that would be clearly incorrect for our use case, so we wanted to flag it immediately with a good error message rather than let something else catch it down the line and have to backtrack. I'm not opposed to removing it.